resalloc_ibm_cloud/powervs/powervs_list_vms.py

```diff
@@ -16,21 +16,40 @@ def list_vms(client: PowerVSClient, pool_id: str) -> set[str]:
 
     Args:
         client: PowerVS client
-        pool_id: Pool ID prefix to filter instances
+        pool_id: Pool ID to filter instances
 
     Returns:
         Set of resource names that match the pool ID
     """
     resources = set()
-
     for instance in client.list_instances():
-        name = instance.get("serverName", "")
-        if name.startswith(pool_id):
-            resources.add(name)
+        instance_name = instance["serverName"]
+        if instance_name.startswith(pool_id):
+            resources.add(instance_name)
 
     return resources
 
 
+def list_volumes_associated_vms(client: PowerVSClient, pool_id: str) -> set[str]:
+    """
+    List all volumes and get their associated VMs.
+
+    Args:
+        client: PowerVS client
+        pool_id: Pool ID to filter VMs
+
+    Returns:
+        Set of VMs names associated with the specified volumes
+    """
+    vms = set()
+    for volume in client.list_volumes():
+        volume_name = volume["volumeID"]
+        if volume_name.startswith(pool_id):
+            vms.add(volume_name.removesuffix("_volume"))
```

Comment:
is there just one volume?

Reply:
... here you include only a volume ending with `_volume`, which might be at most one.

Reply:
If I would be reading the `name = name.rsplit("_", 1)[0]` I'd think we are doing something dynamic/not hardcoded here... but actually it is. The volume name is just the VM name + "_volume", so I lean towards having "_volume" in the code specifically, because if somebody will decide to change the naming and get the error, they will pretty easily spot this bad hardcoded line.

Reply:
> ... here you include only a volume ending with _volume, which might be at most one.

not true, why there would be only one volume ending with such suffix?